subprojects/files/src/test/groovy/org/gradle/api/internal/file/collections/DefaultDirectoryWalkerTest.groovy

```diff
@@ -1,265 +0,0 @@
- * limitations under the License.
- */
-package org.gradle.api.internal.file.collections
-
-import org.gradle.api.file.FileVisitDetails
-import org.gradle.api.file.FileVisitor
-import org.gradle.api.internal.file.TestFiles
-import org.gradle.api.specs.Spec
-import org.gradle.api.tasks.util.PatternSet
-import org.gradle.internal.Factory
-import org.gradle.test.fixtures.AbstractProjectBuilderSpec
-
-class DefaultDirectoryWalkerTest extends AbstractProjectBuilderSpec {
-    def directoryWalkerFactory = new Factory<DirectoryWalker>() {
-        def directoryWalker = new DefaultDirectoryWalker(TestFiles.fileSystem())
-
-        @Override
-        public DirectoryWalker create() {
-            return directoryWalker
-        }
-    }
-    private TestVisitor visitor
-
-    def setup () {
-        visitor = new TestVisitor()
-    }
-
-    def rootDirEmpty() {
-        given:
-        def root = temporaryFolder.createDir("root")
-        def fileTree = new DirectoryFileTree(root, new PatternSet(), directoryWalkerFactory, TestFiles.fileSystem(), false)
-
-        when:
-        fileTree.visit(visitor)
-
-        then:
-        visitor.assertExpectations()
-    }
-
-    def testUsesSpecFromPatternSetToMatchFilesAndDirs() {
-        given:
-        def spec = Mock(Spec)
-        def patternSet = Mock(PatternSet)
-        def fileTree = new DirectoryFileTree(new File("root"), patternSet, directoryWalkerFactory, TestFiles.fileSystem(), false)
-
-        when:
-        fileTree.visit(visitor)
-
-        then:
-        1 * patternSet.getAsSpec() >> spec
-        visitor.assertExpectations()
-    }
-
-    def walkSingleFile() {
-        given:
-        def root = temporaryFolder.createDir("root")
-        def fileToCopy = root.createFile("file.txt")
-        def fileTree = new DirectoryFileTree(fileToCopy, new PatternSet(), directoryWalkerFactory, TestFiles.fileSystem(), false)
-
-        when:
-        visitor.setExpectedVisitations([[fileToCopy]])
-        fileTree.visit(visitor)
-
-        then:
-        visitor.assertExpectations()
-    }
-
-    /*
-    file structure:
-    root
-        rootFile1
-        dir1
-           dirFile1
-           dirFile2
-        rootFile2
-
-        Test that the files are really walked breadth first
-     */
-    def walkBreadthFirst() {
-        given:
-        def root = temporaryFolder.createDir("root")
-        def rootFile1 = root.createFile("rootFile1")
-        def dir1 = root.createDir("dir1")
-        def dirFile1 = dir1.createFile("dirFile1")
-        def dirFile2 = dir1.createFile("dirFile2")
-        def rootFile2 = root.createFile("rootFile2")
-
-        def fileTree = new DirectoryFileTree(root, new PatternSet(), directoryWalkerFactory, TestFiles.fileSystem(), false)
-
-        when:
-        visitor.setExpectedVisitations([[rootFile1, rootFile2], [dir1], [dirFile1, dirFile2]])
-        fileTree.visit(visitor)
-
-        then:
-        visitor.assertExpectations()
-    }
-
-    def walkDepthFirst() {
```

Comment:
I think we should keep these tests and move them to `DefaultDirectoryWalkerTest` maybe?

For example now we don't have any tests for `postfix` traversal any more. Are the other tests covered somewhere else?

Moreover, we need the same tests for `ReproducibleDirectoryWalker`, right?

Reply:
Reverted and fixed, thanks for letting me know they're still relevant.